pkg/selectors/kernel.go

```diff
@@ -769,3 +769,23 @@ func HasOverride(spec *v1alpha1.KProbeSpec) bool {
 	}
 	return false
 }
+
+func HasEarlyBinaryFilter(selectors []v1alpha1.KProbeSelector) bool {
+	if len(selectors) == 0 {
+		return false
+	}
+	for _, s := range selectors {
+		if len(s.MatchPIDs) > 0 ||
+			len(s.MatchNamespaces) > 0 ||
+			len(s.MatchCapabilities) > 0 ||
+			len(s.MatchNamespaceChanges) > 0 ||
+			len(s.MatchCapabilityChanges) > 0 ||
+			len(s.MatchArgs) > 0 {
+			return false
+		}
+	}
+	if len(selectors) == 1 && len(selectors[0].MatchBinaries) > 0 {
```

Comment:
As I understand here, we only apply this optimization in the case where we have a single selector that contains MatchBinaries. Why we cannot use that in the case where we have more than one selector? Maybe add a comment here to explain that.

Reply:
because I need the selector number to do that, so the bpf code will call:
`  match_binaries(&sel_names_map, /* selector index */0);`
I'd need to pass selector indexes somehow..